tsdb/importer/import.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importer
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/go-kit/kit/log"
+	"github.com/otiai10/copy"
+	prom_labels "github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/textparse"
+	"github.com/prometheus/prometheus/tsdb"
+	tsdb_labels "github.com/prometheus/prometheus/tsdb/labels"
+)
+
+// Implementing the error interface to create a
+// constant, which cannot be overridden.
+// https://dave.cheney.net/2016/04/07/constant-errors
+type Error string
+
+func (e Error) Error() string {
+	return string(e)
+}
+
+// This error is thrown when we try to merge/add blocks to an existing TSDB instance,
+// and the new blocks have a time overlap with the current blocks.
+const OverlappingBlocksError = Error("blocks overlap with blocks currently in DB")
+
+// Duration of a block in milliseconds
+const BlockDuration = 2 * 60 * 60 * 1000
+
+type timestamp = int64
+
+type metricSample struct {
+	TimestampMs timestamp
+	Value       float64
+	Labels      tsdb_labels.Labels
+}
+
+// ImportFromFile imports data from a file formatted according to the Prometheus exposition format,
+// converts it into block(s), and places the newly created block(s) in the
+// TSDB DB directory, where it is treated like any other block.
+func ImportFromFile(filePath string, contentType string, dbPath string, skipTimestampCheck bool, logger log.Logger) error {
```

Comment:
don't think  any context should be tied to an existing db
I mean it would be best to pass the `opts.BlockRanges` and just specify where you want the new blocks to be places.

allowing to specify `BlockRanges` removes a lot of magic and leaves it the user to decide how to size the blocks. Can provide the default block ranges as this would be fine for most cases.

something like:
```suggestion
func ImportFromFile(inputFilePath , outputDir,BlockRanges, logger) error {
```

`contentType` - if the tool currently support only one content type don't think we need to pass this as a param 

`skipTimestampCheck` - I did read the arguments about this, but tsdb is now able to handle overlapping blocks so don't think we need this extra logic here. Can add it later if/when needed.

Reply:
In the case of the case passing the current DB path to the function, it allows for the importer to infer the block ranges, and align the new blocks accordingly. I wrote it that way to allow the user to not have to manually pass in the block ranges somehow, and instead just use an existing DB (if it exists), and use its block ranges.

I do agree with the `contentType` change, but I was confused between using the Prometheus expfmt parser, or the OpenMetrics parsers, and that's why I've left it in for now!

Reply:
> I do agree with the contentType change, but I was confused between using the Prometheus expfmt parser, or the OpenMetrics parsers, and that's why I've left it in for now!

you can still have it as a global const. If the tool only supports a single type why do you think it needs to be passed as an arg?

Reply:
I've switched it to being a package level const.